buildSrc/src/main/groovy/org/apache/beam/gradle/VendorJavaPlugin.groovy

```diff
@@ -146,6 +146,8 @@ artifactId=${project.name}
       // Only publish vendored dependencies if specifically requested.
       if (project.hasProperty("vendoredDependenciesOnly")) {
         project.apply plugin: 'maven-publish'
+        // Ensure that we validate the contents of the jar before publishing
+        project.publish.dependsOn project.check
```

Comment:
This will only run `check` on Java gradle projects, not the root project (which include Python, Go, etc.), right?

Reply:
It will only run check on this specific project.